ChopChop/Storage/Models/Recipes/OnlineRecipeRecord.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import FirebaseFirestoreSwift
+
+struct OnlineRecipeRecord {
+    @DocumentID var id: String?
+    var name: String
+    var creator: String
+    var servings: Double
+    @ExplicitNull var cuisine: String?
+    @ExplicitNull var difficulty: Difficulty?
+    var ingredients: [OnlineIngredientRecord]
+    var steps: [String]
+    var stepEdges: [OnlineStepEdgeRecord]
+    var ratings: [RecipeRating] = []
+    @ServerTimestamp var created: Date?
+
+}
+
+extension OnlineRecipeRecord: Equatable {
+}
+
+extension OnlineRecipeRecord: Codable {
+}
+
+enum OnlineRecipeRecordError: Error {
```

Comment:
actually what are these errors for?

Reply:
To catch errors when converting from record to onlinerecipe, cos onlinerecipe requires to have a id and createdDate but the record is optional cos it only gets them when saved to firebase